Sources/Kafka/Configuration/KafkaConfiguration+Metrics.swift

```diff
@@ -0,0 +1,133 @@
+        public var totalBytesReceived: Gauge?
+
+        /// Total number of messages transmitted (produced) to Kafka brokers
+        public var totalMessagesSent: Gauge?
+        /// Total number of message bytes (including framing, such as per-Message framing and MessageSet/batch framing) transmitted to Kafka brokers
+        public var totalMessagesBytesSent: Gauge?
+        /// Total number of messages consumed, not including ignored messages (due to offset, etc), from Kafka brokers.
+        public var totalMessagesRecieved: Gauge?
+        /// Total number of message bytes (including framing) received from Kafka brokers
+        public var totalMessagesBytesRecieved: Gauge?
+
+        /// Number of topics in the metadata cache.
+        public var metadataCacheCount: Gauge?
+
+        private static func record<T: BinaryInteger>(_ value: T?, to: Gauge?) {
+            guard let value,
+                  let to else {
+                return
+            }
+            to.record(value)
+        }
+
+        private static func recordMircoseconds<T: BinaryInteger>(_ value: T?, to: Timer?) {
+            guard let value,
+                  let to else {
+                return
+            }
+            to.recordMicroseconds(value)
+        }
+
+        internal func update(with rdKafkaStatistics: RDKafkaStatistics) {
+            Self.record(rdKafkaStatistics.timestamp, to: self.timestamp)
+            Self.recordMircoseconds(rdKafkaStatistics.time, to: self.time)
+            Self.recordMircoseconds(rdKafkaStatistics.age, to: self.age)
+            Self.record(rdKafkaStatistics.replyQueue, to: self.replyQueue)
+            Self.record(rdKafkaStatistics.messageCount, to: self.messageCount)
+            Self.record(rdKafkaStatistics.messageSize, to: self.messageSize)
+            Self.record(rdKafkaStatistics.messageMax, to: self.messageMax)
+            Self.record(rdKafkaStatistics.messageSizeMax, to: self.messageSizeMax)
+            Self.record(rdKafkaStatistics.totalRequestsSent, to: self.totalRequestsSent)
+            Self.record(rdKafkaStatistics.totalBytesSent, to: self.totalBytesSent)
+
+            Self.record(rdKafkaStatistics.totalResponsesRecieved, to: self.totalResponsesRecieved)
+            Self.record(rdKafkaStatistics.totalBytesReceived, to: self.totalBytesReceived)
+            Self.record(rdKafkaStatistics.totalMessagesSent, to: self.totalMessagesSent)
+            Self.record(rdKafkaStatistics.totalBytesSent, to: self.totalBytesSent)
+
+            Self.record(rdKafkaStatistics.totalMessagesBytesSent, to: self.totalMessagesBytesSent)
+            Self.record(rdKafkaStatistics.totalMessagesRecieved, to: self.totalMessagesRecieved)
+            Self.record(rdKafkaStatistics.totalMessagesBytesRecieved, to: self.totalMessagesBytesRecieved)
+            Self.record(rdKafkaStatistics.metadataCacheCount, to: self.metadataCacheCount)
+        }
+    }
+
+    public enum Metrics: Sendable {
```

Comment:
We don't nee this struct. Let's just make it an optional in the producer and consumer configuration and add the `updateInterval` into the current `KafkaMetrics` struct.

Reply:
Let's do that way